src/manageiq_client/api.py

```diff
@@ -491,20 +491,26 @@ def reload(self):
         self._obj.reload_if_needed()
         reloaded_actions = []
         for action in self._obj._actions:
-            # There can be multiple actions with the same name and different methods
-            # (e.g. actions "delete" with method POST and DELETE).
-            # This makes sure that the attribute refers to the first action and is not redefined
-            # by other action with the same name and different method.
+            action_obj = Action(self, action["name"], action["method"], action["href"])
+            # There can be multiple actions with the same name and different HTTP methods
+            # (e.g. action "delete" with HTTP methods POST or DELETE).
+            # Create action ``.name()`` with default (first) method.
+            # For each method, create action ``.name.METHOD()``.
+            # E.g. default action ``.delete()`` with method POST and actions
+            # ``.delete.POST()`` and ``.delete.DELETE()``.
             if action["name"] not in reloaded_actions:
                 reloaded_actions.append(action["name"])
-                setattr(
-                    self,
-                    action["name"],
-                    Action(self, action["name"], action["method"], action["href"]))
+                setattr(self, action["name"], action_obj)
+            setattr(object.__getattribute__(self, action["name"]),
+                    action["method"].upper(),
+                    action_obj)
 
     def execute_action(self, action_name, *args, **kwargs):
         # To circumvent bad method names, like `import`, you can use this one directly
         action = getattr(self, action_name)
+        action_method = kwargs.pop('action_method', None)
+        if action_method:
+            action = getattr(action, action_method.upper())
```

Comment:
Perhaps drop .upper() here.
If normal way to call is `.action.POST()`, alternative for bad names can also require uppercase: `.execute_action('import', action_method='POST')`.

Also, this is just thin wrapper for `getattr(obj, 'import').POST()`.  That might actually be nicer to write — less different from other methods — than execute_action with action_method?